### PR TITLE
add trial db-logic

### PR DESCRIPTION
**Pull request by dunkleWolke**

### CODE REVIEW

File: app/src/main/java/com/example/banking/Account.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+    constructor() {}
+
+    constructor(name: String, number: String, cardNumber: String) {
+        this.accountName = name
+        this.accountNumber = number
+        this.cardNumber = cardNumber
+    }
+}
```
Comment (Razor0744) [inline app/src/main/java/com/example/banking/Account.kt]: replace to data class kotlin

File: app/src/main/java/com/example/banking/AccountVM.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import android.app.Application
+import androidx.compose.runtime.mutableStateListOf
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.room.Room
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class AccountVM (application: Application) : AndroidViewModel(application) {
```
Comment (Razor0744) [inline app/src/main/java/com/example/banking/AccountVM.kt]: Android View Model? DB in view model? change all code, use koin and provide application there

File: app/src/main/java/com/example/banking/Transaction.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    constructor(accountId: Int, name: String, number: String, date: String, status: String, amount: String) {
+        this.accountId = accountId
+        this.name = name
+        this.number = number
+        this.date = date
+        this.status = status
+        this.amount = amount
+    }
+}
```
Comment (Razor0744) [inline app/src/main/java/com/example/banking/Transaction.kt]: the same "data class Kotlin"

File: app/src/main/java/com/example/banking/Account.kt (hunk 1 above)
Comment (dunkleWolke) [reply app/src/main/java/com/example/banking/Account.kt]: fix

File: app/src/main/java/com/example/banking/AccountVM.kt (hunk 2 above)
Comment (dunkleWolke) [reply app/src/main/java/com/example/banking/AccountVM.kt]: I know a little bit about koin, but I tried to do it

File: app/src/main/java/com/example/banking/Transaction.kt (hunk 3 above)
Comment (dunkleWolke) [reply app/src/main/java/com/example/banking/Transaction.kt]: fix

File: app/src/main/java/com/example/banking/AppDB.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        fun getDB(context: Context): AppDB {
+            return INSTANCE ?: synchronized(this){
+                val instance = Room.databaseBuilder(
+                    context.applicationContext,
+                    AppDB::class.java,
+                    "m-banking",
+                ).build()
+                INSTANCE = instance
+                return instance
+            }
```
Comment (Razor0744) [inline app/src/main/java/com/example/banking/AppDB.kt]: If get instance you don't need to use companion object

File: app/src/main/java/com/example/banking/MainActivity.kt [hunk 5]
```diff
@@ -11,37 +11,38 @@ import androidx.compose.material3.Text
...
+
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         enableEdgeToEdge()
+        startKoin {
+            androidContext(this@MainActivity)
+            modules(appModule)
+        }
+
```
Comment (Razor0744) [inline app/src/main/java/com/example/banking/MainActivity.kt]: Use koin with application

File: app/src/main/java/com/example/banking/MainActivity.kt [hunk 6]
```diff
@@ -11,37 +11,38 @@ import androidx.compose.material3.Text
...
-}
+val appModule = module {
 
-@Preview(showBackground = true)
-@Composable
-fun GreetingPreview() {
-    BankingTheme {
-        Greeting("Android")
-    }
+    single { AppDB.getDB(get()).accountDao() }
```
Comment (Razor0744) [inline app/src/main/java/com/example/banking/MainActivity.kt]: You can create instance of db in the koin

**Pull request merged**